proxy/proxy.go

```diff
@@ -135,6 +135,12 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// If the request is a gRPC request, we need to set the Content-Type
+	// header to application/grpc.
+	if strings.HasPrefix(r.Header.Get(hdrContentType), hdrTypeGrpc) {
```

Comment:
Hmm, this might fix a long-standing error that we see sometimes. I just hope it doesn't introduce any side effects (I remember the whole header handling for gRPC being super brittle with many weird behaviors).